core/arch/arm/kernel/unwind_arm32.c

```diff
@@ -123,18 +116,9 @@ static struct unwind_idx *find_index(uint32_t addr, uaddr_t exidx,
 	int32_t prel31_addr;
 	vaddr_t func_addr;
 
-	if (exidx && exidx_sz) {
-		/* Unwinding a user TA */
-		start = (struct unwind_idx *)exidx;
-		idx_start = (vaddr_t)exidx;
-		idx_end = (vaddr_t)(exidx + exidx_sz);
-
-	} else {
-		/* Unwinding the kernel stack: use kernel tables */
-		start = &__exidx_start;
-		idx_start = (vaddr_t)&__exidx_start;
-		idx_end = (vaddr_t)&__exidx_end;
-	}
+	start = (struct unwind_idx *)exidx;
+	idx_start = (vaddr_t)exidx;
```

Comment:
Perhaps the type of `exidx` should be `vaddr_t` instead to avoid this cast and the cast below.